visualisation/nzvm.py

```diff
@@ -130,14 +135,34 @@ def main():
     if length != nx*ny*nz:
         print("Error: Inconsistent VM size. Check model and nx, ny, nz parameters and try again. Expected",nx*ny*nz,"values, got",len(array))
         quit()
+
+    if args.perturbation:
+        pert_length, pert_array = load_data(args.perturbation, nx, ny, nz)
+        if pert_length != nx*ny*nz:
+            print("Error: Inconsistent perturbation file size. Check model and nx, ny, nz parameters and try again. Expected",nx*ny*nz,"values, got",len(pert_array))
+            quit()
     
     if axis == 'x':
         data = array[start_point[1],:,start_point[0]:(start_point[0]+delta+1)] * KM_TO_M
     else:
         data = array[start_point[1]:(start_point[1]+delta+1),:,start_point[0]] * KM_TO_M
         data = np.swapaxes(data,0,1)
-        
-    display_data_matplot(data, int(len(data)/nz), nz, args.maximum, hh)
+    
+    if args.perturbation:
+        fig, axs = plt.subplots(2, 1, constrained_layout=True)
+        display_data_matplot(data, args.maximum, hh, axs[0], "Cross section (without perturbation)")
+
+        pert_mixed_array = array * pert_array
+        if axis == 'x':
+            pert_data = pert_mixed_array[start_point[1],:,start_point[0]:(start_point[0]+delta+1)] * KM_TO_M
```

Comment:
Just because this line is getting unweildly. I think you'd be better off setting your some of your variables in this branch and then doing the slicing generically after the conditional block. (transpose may help here too)

Unsure if you have a better idea on how to tidy it up a little.